jupyter_accessibility/project.py

```diff
@@ -0,0 +1,428 @@
+            data = loads(package.read_text(encoding="utf-8"))
+            name = data["name"]
+            if name not in visited:
+                visited.add(name)
+                yield dict(
+                    basename="symlink",
+                    name=name,
+                    file_dep=list(deps) + [package],
+                    actions=[do(f"yarn link --link-folder {self.links}", cwd=package.parent)],
+                    targets=[self.links / name / "package.json"],
+                )
+
+    def close_yarn_symlinks(self, target):
+        """created symbollic links to local development dependencies created by new_symlinks"""
+
+        visited = set()
+        for package in self.get_packages():
+            data = loads(package.read_text(encoding="utf-8"))
+            name = data["name"]
+            if name not in visited:
+                visited.add(name)
+                # should use basename
+                yield dict(
+                    basename="install",
+                    name=f"yarn:link:close:{name}",
+                    file_dep=[package],
+                    actions=[do(f"yarn link --link-folder {self.links} {name}", cwd=target.path)],
+                    targets=[target.links / name / "package.json"],
+                )
+
+    def pip_install(self):
+        """pip install the python package"""
+        yield dict(
+            basename="install",
+            name=f"pip:{self.name}",
+            actions=[
+                do(f"{self.conda} python -m pip install jupyter_packaging"),
+                do(
+                    f"{self.conda} python -m pip install -e.",  # many settings are held in the env
+                    cwd=self.path,
+                    env=PIP_BUILD_ENV,
+                ),
+            ],
+        )
+
+    def tasks(self):
+        yield self.clone()
+        # we need to clone before we can do anything
+        # these have to happen afterwards. the order in the method has no bearing on the execution order
+        yield self.yarn_install()
+
+    def get_repo(self, id):
+        if isinstance(id, type):
+            id = id.id
+
+        return self.parent.repos[id]
+
+
+# documentation tests and unit tests would have different task specifications
+class PlayWrightTests:
+    """a mixin with playwright tests"""
+
+    def tests(self):
+        target = self.dir / AXE_TEMPLATE.name
+        yield dict(
+            basename="test_setup",
+            name="copy-templates",
+            actions=[(mv, [AXE_TEMPLATE, target])],
+            clean=[(rmdir, [target])],
+            uptodate=[target.exists()],
+        )
+        yield dict(
+            basename="test_setup",
+            name="pip",
+            actions=[do(f"{self.conda} pip install -r requirements.txt", cwd=target)],
+        )
+        yield dict(
+            basename="test_setup",
+            name="yarn",
+            actions=[do(f"{self.conda} yarn install", cwd=target)],
+        )
+        if CI or GITPOD:
```

Comment:
I am not sure how playwright caching works? maybe @gabalafou does - but  for now this will be our starting point

Reply:
yea.... this is going to take a meeting of the minds. there should be a way to inspect the state of their cache. might take a little pair programming.

Reply:
Check out [caching browsers](https://playwright.dev/docs/ci#caching-browsers) in the Playwright docs